### PR TITLE
docs: RootTheme, themeMerge

### PR DESCRIPTION
**Pull request by zero0Halo**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

* Adds documentation on `RootTheme`
* Adds documentation on `themeMerge`
* Removes all documentation on `RootTheme` from the `ThemeProvider` page as it was very sad compared to what is introduced above
* Added a component, `ThemeExplorer` to help visualize themes in examples
* Removed the `colors` page as it is no longer relevant

### CODE REVIEW

File: docs/src/pages/theme/root-theme.mdx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+## Breakdown
+
+Now that we know how to use `RootTheme` (or a custom theme based on it), here is a breakdown of
+its key/values.
+
+---
+
+#### borders
+The CSS properties, `border`, `border-top`, `border-right`, `border-bottom`, and `border-left`
```
Comment (jcristol) [inline docs/src/pages/theme/root-theme.mdx:111]: This is a confusing sentence to me. I would like it if it followed the format in the sentence `The CSS properties, color, background-color and border-color can all reference the keys in colors.` (the CSS properties `border`, `border-top`, `border-right`, `border-bottom`, and `border-left`, ''can all reference the keys in borders)

File: docs/src/pages/theme/root-theme.mdx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+The CSS properties, `color`, `background-color` and `border-color` can all reference the keys in
+`colors`.
+
+<ThemeExplorer openTo="colors" />
+
+---
+
+#### fonts
+The CSS property, `font-family` associates to these keys.
```
Comment (jcristol) [inline docs/src/pages/theme/root-theme.mdx]: same as may previous comment about confusing sentence.

File: docs/src/pages/theme/root-theme.mdx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+<ThemeExplorer openTo="fonts" />
+
+---
+
+#### radii
+The CSS property, `border-radius` associates to these keys.
```
Comment (jcristol) [inline docs/src/pages/theme/root-theme.mdx]: same as may previous comment about confusing sentence.

File: docs/src/pages/theme/theme-merge.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import { Link } from 'gatsby';
+import { ThemeExplorer } from '../../components/ThemeExplorer';
+
+# themeMerge()
```
Comment (jcristol) [inline docs/src/pages/theme/theme-merge.mdx:4]: this is probably to late for this. But i don't love the function name `themeMerge`. I view merges as forming the union of two mostly disjoint sets, with undefined behavior for what happens to the set of things in the overlap. This function in my head should be called `themeEditor` or `themeSetter`. We take in a set of settings to override the default root theme configuration.

File: docs/src/pages/theme/root-theme.mdx (hunk 1 above)
Comment (zero0Halo) [reply docs/src/pages/theme/root-theme.mdx:111]: It's a slight change, but a good one. I'll update the wording in this and the other instances.

File: docs/src/pages/theme/theme-merge.mdx (hunk 4 above)
Comment (zero0Halo) [reply docs/src/pages/theme/theme-merge.mdx:4]: For the most part, `themeMerge` is an object merge utility that defaults to using `RootTheme`. Because it follows similar patterns as other merge utilities I chose its name to reflect that. I feel like it could go either way, but I lean more towards 'merge' for that reason.

File: docs/src/pages/theme/root-theme.mdx (hunk 1 above)
Comment (jcristol) [reply docs/src/pages/theme/root-theme.mdx:111]: woot

File: docs/src/pages/theme/theme-merge.mdx (hunk 4 above)
Comment (jcristol) [reply docs/src/pages/theme/theme-merge.mdx:4]: kewl beans

File: docs/src/components/ThemeExplorer/DataBlock/DataBlock.component.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    return (
+        <span>
+            {typeof data === 'function' ? (
+                // This is a little hacky. This is most likely true, that css() is being used if the
+                // type is function. It is truthy as of right now (v1.3.7)
+                <a
+                    href="https://xstyled.dev/docs/migrate-from-styled-components/#css-utility-is-required"
+                    target="_blank"
+                >
+                    css()
```
Comment (brandonlilly) [inline docs/src/components/ThemeExplorer/DataBlock/DataBlock.component.tsx]: Mmm this won't hold up forever. Should we just say that its a function and future proof it now?

File: docs/src/pages/theme/theme-merge.mdx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+`themeMerge` is a utility function which returns a deep cloned object based on a settings object
+passed by the user along with a base theme object. If no base theme object is provided, `themeMerge`
+defaults to **Anchor**'s <Link to="/theme/root-theme">`RootTheme`</Link>.
+
+**Arguments**
+
+* `object`: The settings the user wishes to merge with a base theme object.
```
Comment (brandonlilly) [inline docs/src/pages/theme/theme-merge.mdx]: seems a bit weird to just have it labelled as `object` without a name for the arguments.

`settings` - `object`  - The settings the user wishes to merge with a base theme object.  
`rootTheme` - object` - ...

?

File: docs/src/pages/theme/theme-merge.mdx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+It is also possible to remove fields from the base theme object. This can be useful in some situations
+where a default style is difficult to work around. This is intended to be used to remove a
+single css property, but it is not limited to that. Properties whose values are objects along with
+their children can also be removed, though none of the root properties can be. In the example below
+the `mono` property is being removed from the theme.
+
+Try changing `mono` to `base`, and see in the resulting example how `mono` returns and now `base` is
+removed.
+
+```tsx live
```
Comment (brandonlilly) [inline docs/src/pages/theme/theme-merge.mdx:66]: I like the live example

**Review by jcristol — APPROVED** [review]

File: docs/src/components/ThemeExplorer/DataBlock/DataBlock.component.tsx (hunk 5 above)
Comment (zero0Halo) [reply docs/src/components/ThemeExplorer/DataBlock/DataBlock.component.tsx]: We had talked about what to do here and agreed that perhaps it was enough to let the user know `css()` was being used, though not technically correct. I can put that it's a function instead to future-proof. I'm concerned that it'll lead to trying to explain to users what's going on with `Button`.

Comment (zero0Halo) [reply docs/src/components/ThemeExplorer/DataBlock/DataBlock.component.tsx]: Bah, stupid ctrl+enter commented rather than going to a new line. Stupid different RTE's. :)

I will take a stab at showing `func()` instead with a brief explanation as to how `Button` uses it.

File: docs/src/pages/theme/theme-merge.mdx (hunk 6 above)
Comment (zero0Halo) [reply docs/src/pages/theme/theme-merge.mdx]: Yeah, probably right there. Putting in the interface here could be messy, but it's better than nothing.

**Review by brandonlilly — APPROVED** [review]
👍
